app/src/store/stores/subServerStore.ts

```diff
@@ -0,0 +1,51 @@
+import { makeAutoObservable, runInAction } from 'mobx';
+import { Store } from 'store';
+import * as STATUS from 'types/generated/lit-status_pb';
+
+interface SubServers {
+  [key: string]: STATUS.SubServerStatus.AsObject;
+}
+
+export default class SubServerStore {
+  private _store: Store;
+
+  loading = false;
+
+  /** an array map of available subservers and their current status */
+  serverStatus: STATUS.SubServerStatusResp.AsObject = { subServersMap: [] };
```

Comment:
I defined `SubServerStatus` like you suggested and put it in `types/state.ts`. I refactored all of the subserver logic to use this type instead of storing the RPC response object.